src/controllers/connectionManager.ts

```diff
@@ -788,6 +788,9 @@ export default class ConnectionManager {
 			title: LocalizedConstants.connectProgressNoticationTitle,
 			cancellable: false
 		}, async (_progress, _token) => {
+			if (!connectionCreds.server) {
```

Comment:
Why is connect even being called? The user is still choosing a connection at that point, so it shouldn't even be trying to connect to anything.

Reply:
Looking at the code (but not slapping a debugger on it), I'm guessing the callstack is:

```
DatabaseProject:
createProjectFromDatabaseQuickpick.ts
  createNewProjectFromDatabaseWithQuickpick() ->
  vscodeMssqlApi.promptForConnection() ->

Vscode-Mssql:
connectionUI.ts
  promptForConnection() ->
  handleSelectedConnection() ->
  createAndSaveProfile() ->
  validateAndSaveProfile()
```

Which would attempt to connect to connect to anything in the dropdown that gets highlighted, I believe.

Reply:
Not exactly, as first 'connect' is always called (this is where I'm validating server now) and 'err' is expected that triggers promptForConnection:

https://github.com/microsoft/azuredatastudio/blob/0d8e1882fa7f974b27bc4b68fe6203e4a627aae3/extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseQuickpick.ts#L35C1-L46

```ts
// Get the list of databases now to validate that the connection is valid and re-prompt them if it isn't
try {
	connectionUri = await vscodeMssqlApi.connect(connectionProfile);
	dbs = (await vscodeMssqlApi.listDatabases(connectionUri))
		.filter(db => !constants.systemDbs.includes(db)); // Filter out system dbs
} catch (err) {
	// The mssql extension handles showing the error to the user. Prompt the user
	// for a new connection and then go and try getting the DBs again
	connectionProfile = await vscodeMssqlApi.promptForConnection(true);
	if (!connectionProfile) {
		// User cancelled
		return undefined;
	}

}
```

Reply:
Fixing in vscode-mssql is eitherway a good thing I guess, to ensure servername is not empty before reaching STS.